src/libostree/ostree-repo.c

```diff
@@ -1928,17 +1927,15 @@ ostree_repo_remote_list (OstreeRepo *self,
 
   _ostree_repo_remote_list (self, remotes_ht);
 
-  n_remotes = g_hash_table_size (remotes_ht);
-
+  const guint n_remotes = g_hash_table_size (remotes_ht);
   if (n_remotes > 0)
     {
-      GList *list, *link;
       guint ii = 0;
 
       remotes = g_new (char *, n_remotes + 1);
 
-      list = g_hash_table_get_keys (remotes_ht);
-      list = g_list_sort (list, (GCompareFunc) strcmp);
+      GList *list = g_hash_table_get_keys (remotes_ht);
+      GList *link = g_list_sort (list, (GCompareFunc) strcmp);
```

Comment:
I think this changes the actual logic. The for-loop below seems to rely on `list`, which should be sorted at that point, while `link` is set up directly in the init step.

Reply:
Thanks, I just dropped that commit; it was mostly to test LGTM.com's C/C++ handling which as it turns out is less useful than I thought.